packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts

```diff
@@ -1,44 +1,41 @@
 import { Injectable } from '@nestjs/common';
-import { lastValueFrom } from 'rxjs';
 import {
   JobsDiscoveryParamsCommand,
   JobsDiscoveryParamsData,
   JobsDiscoveryResponse,
 } from './interfaces/jobs-discovery.interface';
-import { HttpService } from '@nestjs/axios';
 import { InjectMapper } from '@automapper/nestjs';
 import { Mapper } from '@automapper/core';
+import { CommonHttpUtilService } from '../../common/utils/common-http-util.service';
+import { AxiosRequestConfig } from 'axios';
 import { RequestContext } from '../../common/utils/request-context.util';
 
 @Injectable()
 export class JobsDiscoveryService {
   constructor(
-    public httpService: HttpService,
     @InjectMapper() private readonly mapper: Mapper,
+    private readonly httpService: CommonHttpUtilService,
     private readonly requestContext: RequestContext,
   ) {}
 
   async processJobsDiscovery(
-    jobsDiscoveryParamsCommand: JobsDiscoveryParamsCommand,
+    command: JobsDiscoveryParamsCommand,
   ): Promise<JobsDiscoveryResponse> {
-    const jobsDiscoveryParamsData = this.mapper.map(
-      jobsDiscoveryParamsCommand,
+    const url = command.exchange_oracle_url;
+    const token = this.requestContext.token;
+    const data = this.mapper.map(
+      command,
       JobsDiscoveryParamsCommand,
       JobsDiscoveryParamsData,
     );
-    try {
-      const url = jobsDiscoveryParamsCommand.exchange_oracle_url;
-      const token = this.requestContext.token;
-      const options = {
-        method: 'GET',
-        url: `${url}/jobs`,
-        params: jobsDiscoveryParamsData,
-        headers: { Authorization: `Bearer ${token}` },
-      };
-      const response = await lastValueFrom(this.httpService.request(options));
-      return response.data;
-    } catch (error) {
-      throw error;
-    }
+    const options: AxiosRequestConfig = {
+      method: 'GET',
+      url: `${url}/jobs`,
+      params: data,
+      headers: { Authorization: `Bearer ${token}` },
+    };
+    return this.httpService.callExternalHttpUtilRequest<JobsDiscoveryResponse>(
+      options,
```

Comment:
IMO we shouldn't call an external service in the service layer, it could be behind an abstraction